### PR TITLE
[MPS] Map Conv to Depthwise when input_channels==same as num_groups.

### PR DESCRIPTION
**Pull request by kulinseth**
This helps with performance in MobileNet, NasNet etc.

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Convolution.mm [hunk 1]
```diff
@@ -12,6 +12,24 @@
...
 namespace native {
 
+void fill_depthwise_conv_desc(MPSGraphDepthwiseConvolution3DOpDescriptor* descriptor_,
+                    NSUInteger strideInX, NSUInteger strideInY,
+                    NSUInteger dilationRateInX, NSUInteger dilationRateInY,
+                    NSUInteger paddingHorizontal, NSUInteger paddingVertical,
+                    c10::MemoryFormat memory_format, NSUInteger groups) {
+  descriptor_.strides = @[@1, [[NSNumber alloc] initWithInteger: strideInY],
+                              [[NSNumber alloc] initWithInteger: strideInX]];
+  descriptor_.dilationRates = @[@1, [[NSNumber alloc] initWithInteger: dilationRateInY],
```
Comment (razarmehr) [inline aten/src/ATen/native/mps/operations/Convolution.mm:22]: For better readability, can we replace all of these allocations of `[NSNumber alloc] initWithInteger: X` with just `@X`? Like this one:
`descriptor_.strides = @[@1, @(strideInY), @(strideInX)];`

**Review by razarmehr — COMMENTED** [review]
Can we add a test case in test_mps for this new feature?

File: aten/src/ATen/native/mps/operations/Convolution.mm (hunk 1 above)
Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/Convolution.mm:22]: This is for existing test cases.. we remap the existing Conv to depth wise for improved performance. It already has bunch of conv tests.

Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/Convolution.mm:22]: The cleanup I can do while upstreaming..

**Review by DenisVieriu97 — APPROVED** [review]
looks good

**Pull request merged**